client/src/main/java/org/apache/celeborn/client/ReviveManager.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.celeborn.client;
+
+import java.util.*;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.apache.celeborn.common.CelebornConf;
+import org.apache.celeborn.common.protocol.PartitionLocation;
+import org.apache.celeborn.common.protocol.ReviveRequest;
+import org.apache.celeborn.common.protocol.message.StatusCode;
+import org.apache.celeborn.common.util.ThreadUtils;
+
+class ReviveManager {
+  private static final Logger logger = LoggerFactory.getLogger(ShuffleClientImpl.class);
+
+  LinkedBlockingQueue<ReviveRequest> requestQueue = new LinkedBlockingQueue<>();
+  private final long interval;
+  private final int batchSize;
+  ShuffleClientImpl shuffleClient;
+  private ScheduledExecutorService batchReviveRequestScheduler =
+      ThreadUtils.newDaemonSingleThreadScheduledExecutor("batch-revive-scheduler");
+
+  public ReviveManager(ShuffleClientImpl shuffleClient, CelebornConf conf) {
+    this.shuffleClient = shuffleClient;
+    this.interval = conf.clientPushReviveInterval();
+    this.batchSize = conf.clientPushReviveBatchSize();
+
+    batchReviveRequestScheduler.scheduleAtFixedRate(
+        () -> {
+          Map<Integer, Set<ReviveRequest>> shuffleMap = new HashMap<>();
+          do {
+            ArrayList<ReviveRequest> batchRequests = new ArrayList<>();
+            requestQueue.drainTo(batchRequests, batchSize);
+            batchRequests.forEach(
+                (req) -> {
+                  Set<ReviveRequest> set =
+                      shuffleMap.computeIfAbsent(req.shuffleId, id -> new HashSet<>());
+                  set.add(req);
+                });
```

Comment:
The `forEach` is neither readable nor efficient here.
```suggestion
            for (ReviveRequest req in batchRequests) {
              Set<ReviveRequest> set =
                  shuffleMap.computeIfAbsent(req.shuffleId, id -> new HashSet<>());
              set.add(req);
            }
```